### PR TITLE
Fix cl2.hpp so it compiles for both 1.2 and 2.0

### PR DESCRIPTION
**Pull request by ouakheli**
The compiler complains about missing braces when compiling for 2.0, and
complains about "too many braces" for 1.2, because cl_image_desc has a
union in 2.0, but just a cl_mem in 1.2.

Set only needed members of cl_image_desc struct.

### CODE REVIEW

**Comment by bashbaug** [comment]
Thank you for your contribution!  The travis build failure is not caused by your change - see #129.

**Review by bashbaug — APPROVED** [review]
LGTM, this is a nice improvement.

Do we need to enhance our CI coverage to catch issues like this in the future?

It'd be great to get another quick review from @jrprice or @kpet  before merging.

**Review by jrprice — APPROVED** [review]

**Comment by jrprice** [comment]
> Do we need to enhance our CI coverage to catch issues like this in the future?

Yes, wouldn't hurt. I've opened https://github.com/KhronosGroup/OpenCL-CLHPP/issues/133 for this.

**Pull request merged**
